docs/guides/server/keycloak-truststore.adoc

```diff
@@ -2,20 +2,35 @@
+When {project_name} communicates with external services or has an incoming connection through TLS, it has to validate the remote certificate in order to ensure it is connecting to a trusted server. This is necessary in order to prevent man-in-the-middle attacks.
 
-The truststore is used when connecting securely to identity brokers, LDAP identity providers, when sending emails, and for backchannel communication with client applications. It is also useful
-when you want to change the policy on how host names are verified and trusted by the server.
+The certificates of these clients or servers, or the CA that signed these certificates, must be put in a truststore. This truststore is then configured for use by Keycloak.
 
-By default, a truststore provider is not configured, and any TLS/HTTPS connections fall back to standard Java Truststore configuration. If there is no trust established, then these outgoing requests will fail.
+== Configuring the Global Truststore
 
-== Configuring the {project_name} Truststore
+You may use the --truststore-paths option to create a single merged truststore, which will include the existing Java default truststore certs. That merged truststore will be used as the system default truststore via the javax.net.ssl properties, and as the default spi-truststore.  For example:
 
-You can add your truststore configuration by entering this command:
+<@kc.start parameters="--truststore-paths=myTrustStore.pfx,myOtherTrustStore.pem" />
+
+The truststores must be PEM files, or use the .pfx or .p12 extension and be in PKCS12 format.  If in PKCS12, the certs must be unencrypted - meaning no password is expected.  
+
+It is still possible to directly set your own javax.net.ssl truststore System properties, but it's recommended to use the --truststore-paths instead.
+
+== Configuring the SPI Truststore
+
+The SPI truststore file option configures a truststore to securely communicate with identity brokers, LDAP identity providers, when sending emails, for backchannel communication with client applications, and some other certificate verification. This is a subset of what the global truststore is used for.
+
+If the SPI truststore file option is not specified, and a global truststore is configured, the global truststore will be used as the SPI truststore.
+
+The SPI truststore file option will supersede whatever is set as the global truststore.
+
+This configuration is also useful when you want to change the policy on how host names are verified and trusted by the server.  Note that hostname-verification-policy may be set independently of the other properties.
+
+For example you may configure an SPI Trustore with the following command:
```

Comment:
Huh? I don't get it. The `SPI Truststore` is or at least was the global truststore for the Keycloak server, and now we have another global truststore.

Reply:
We could possibly remove all mentions of the SPI truststore whatsoever and limit it to just `hostname-verification-policy`. @stianst @shawkins WDYT?

Reply:
> Huh? I don't get it. The SPI Truststore is or at least was the global truststore for the Keycloak server, and now we have another global truststore.

Unfortunately the SPI Truststore was not truly global - it did not take the place of the javax.net.ssl.trustStore, which could still be needed by drivers or custom code paths.

> We could possibly remove all mentions of the SPI truststore whatsoever and limit it to just hostname-verification-policy. @stianst @shawkins WDYT?

Would it make sense to just alias the hostname-verification-policy property as something else so that the entire set of exsiting spi-truststore-file properties can be deprecated / hidden?

Reply:
Removed the doc reference about the spi-truststore-file.  A hostname-verification-policy property was added that maps to kc.spi-truststore-file-hostname-verification-policy